lib/injectable/dependency.rb

```diff
@@ -2,14 +2,30 @@ module Injectable
   # Initialize a dependency based on the options or the block passed
   Dependency = Struct.new(:name, :block, :class, :call, :with, :depends_on, keyword_init: true) do
     def instance(args: [], namespace: nil)
-      args = wrap_args(args)
-      wrap_call build_instance(args, namespace: namespace)
+      positional_args, kwargs = split_args(args)
+
+      wrap_call build_instance(positional_args, kwargs, namespace: namespace)
     end
 
     private
 
-    def wrap_args(args)
+    def split_args(args)
+      positional_args = []
+      kwargs = {}
+
+      preprocess_args(args).each do |arg|
+        arg.is_a?(Hash) ? kwargs.merge!(arg) : positional_args << arg
```

Comment:
It probably makes sense to check ActiveSupport implementation of `extract_options` which has exactly the same purpose.

Reply:
Seems like it just checks if the last element is a hash

```rb
# File activesupport/lib/active_support/core_ext/array/extract_options.rb, line 24
  def extract_options!
    if last.is_a?(Hash) && last.extractable_options?
      pop
    else
      {}
    end
  end

# File activesupport/lib/active_support/core_ext/array/extract_options.rb, line 9
  def extractable_options?
    instance_of?(Hash)
  end
```